### ISSUE TITLE
Users Page Stuck on Loading After Updating to Version 26.0.7 When User-Based Policies Are Configured

### ISSUE THREAD

**Issue by amiramirifar2** [keycloak/keycloak#35990]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/fine-grained-permissions

### Describe the bug

I created several users, each of whom is an administrator of a separate group.  
Each user has the `query-users` and `query-groups` roles from the **realm-management roles list**.  

Additionally, I created **user-based policies** in the **realm-management authorization tab** for each user and assigned those policies to the following permissions of their respective groups:  
- `manage-member`  
- `manage-membership`  
- `view`  
- `view-members`  

Until version **25.x**, each administrator could view and manage users within their group.  
However, after updating to version **26.0.7**, the **users page** remains stuck on loading and fails to display any users.

### Version

26.0.7

### Regression

- [ ] The issue is a regression

### Expected behavior

view and manage the users in the group

### Actual behavior

not show user list

### How to Reproduce?

1. **Setup Users and Groups**:  
   - Create multiple users, where each user is an administrator of a separate group.  
     - Example:  
       - `User1` → Administrator of `GroupA`  
       - `User2` → Administrator of `GroupB`  
       - `User3` → Administrator of `GroupC`  

2. **Assign Roles**:  
   - Assign the following **realm-management roles** to each user:  
     - `query-users`  
     - `query-groups`  

3. **Configure User-Based Policies**:  
   - In the **realm-management authorization tab**, create a **user-based policy** for each user.  
   - Assign created policy to the following permissions of the corresponding group:  
     - `manage-member`  
     - `manage-membership`  
     - `view`  
     - `view-members`  
   - Example:  
     - `User1` → Permissions for `GroupA`  
     - `User2` → Permissions for `GroupB`  
     - `User3` → Permissions for `GroupC`  

4. **Test Behavior**:  
   - **Before Updating (Version 25.x):**  
     - Each user can access the **Users page** and successfully view and manage users in their group.  
   - **After Updating (Version 26.0.7):**  
     - When any user tries to access the **Users page**, the page remains stuck on **loading** and fails to display the user list.  

### Anything else?

_this problem is not for realm admins_

**Comment by pedroigor** [comment]
@amiramirifar2 Are you able to share a thread dump when waiting for the server response?

**Comment by amiramirifar2** [comment]
Could you tell me which part needs further explanation?

**Comment by amiramirifar2** [comment]
> @amiramirifar2 Are you able to share a thread dump when waiting for the server response?

could you guide me how to export a dump?

**Comment by amiramirifar2** [comment]
> Thanks for reporting this issue, but there is insufficient information or lack of steps to reproduce.
> 
> Please provide additional details, otherwise this issue will be automatically closed within 14 days.

I update issue

**Comment by pedroigor** [comment]
Please, try `kill -3 <pid>`.

**Comment by amiramirifar2** [comment]
> Please, try `kill -3 <pid>`.

there is no output after running this command

i run keycloak i kubernetes

**Comment by pedroigor** [comment]
@amiramirifar2 You need to shell into the container and run `kill -3 1`. Within the container, the server process should be running with PID `1`.

In any case, I tried to reproduce it locally by following your steps. I was not able to see the behaviour you are reporting with only those 3 users. I guess you have more users in the realm and have them assigned to the different groups. If you can provide more details about this too, I appreciate.

**Comment by amiramirifar2** [comment]
> @amiramirifar2 You need to shell into the container and run `kill -3 1`. Within the container, the server process should be running with PID `1`.
> 
> In any case, I tried to reproduce it locally by following your steps. I was not able to see the behaviour you are reporting with only those 3 users. I guess you have more users in the realm and have them assigned to the different groups. If you can provide more details about this too, I appreciate.

yes i have about  20000 and 1 group in root, 200 group in level 2 and each level 2 group has 4 child in level 3 and my users are grouped into various groups at these 3 levels, with group sizes ranging from 5 to 50 members.

**Comment by pedroigor** [comment]
@amiramirifar2 Thanks for the details. The output of that command should go to the system out or any log file you have configured. You should be ble to grab it from the container using `kubectl logs`. 

In a regular container and using docker, you can see the output if running `docker logs`. 

In the meantime, I will try to use the https://github.com/keycloak/keycloak-benchmark to have a similar database. But it would be nice if you can provide the thread dump as it should be quicker and (hopefully) easier to spot where the server is stuck. Please, make sure to take the thread dump at the moment you see the server hanging.

**Comment by amiramirifar2** [comment]
> @amiramirifar2 Thanks for the details. The output of that command should go to the system out or any log file you have configured. You should be ble to grab it from the container using `kubectl logs`.
> 
> In a regular container and using docker, you can see the output if running `docker logs`.
> 
> In the meantime, I will try to use the https://github.com/keycloak/keycloak-benchmark to have a similar database. But it would be nice if you can provide the thread dump as it should be quicker and (hopefully) easier to spot where the server is stuck. Please, make sure to take the thread dump at the moment you see the server hanging.

I am trying to get the output.
Should I run this command while the user list is being loaded?

**Comment by pedroigor** [comment]
Yes, please.

**Comment by ahus1** [comment]
This is reaching out to users of fine-grained-admin-permissions. 

The Keycloak team is working on V2 of the feature which is planned to be part of KC 26.2, and we're posting updates and looking for feedback in the following discussion: https://github.com/keycloak/keycloak/discussions/37133

It would be good to know if you are willing to try it out, and if you could subscribe to that discussion if you are interested in receiving updates to that feature. Thanks!
